datafusion/physical-expr-common/src/stats_v2.rs

```diff
@@ -0,0 +1,887 @@
+            Bernoulli { p } => is_valid_bernoulli(p),
+            Unknown {
+                mean,
+                median,
+                variance,
+                range,
+            } => is_valid_unknown(mean, median, variance, range),
+            Uniform { .. } => Ok(true),
+        }
+    }
+
+    /// Extracts the mean value of the given statistic, depending on its distribution:
+    /// - A [`Uniform`] distribution's interval determines its mean value, which is the
+    ///   arithmetic average of the interval endpoints.
+    /// - An [`Exponential`] distribution's mean is calculable by formula `offset + 1/λ`,
+    ///   where λ is the (non-negative) rate.
+    /// - A [`Gaussian`] distribution contains the mean explicitly.
+    /// - A [`Bernoulli`] distribution's mean is equal to its success probability `p`.
+    /// - An [`Unknown`] distribution _may_ have it explicitly, or this information may
+    ///   be absent.
+    pub fn mean(&self) -> Result<ScalarValue> {
+        match &self {
+            Uniform { interval } => {
+                let agg = interval
+                    .lower()
+                    .add_checked(interval.upper())?
+                    .cast_to(&DataType::Float64)?;
+                agg.div(ScalarValue::Float64(Some(2.)))
+            }
+            Exponential {
+                rate,
+                offset,
+                positive_tail,
+            } => {
+                let one = ScalarValue::new_one(&offset.data_type())?;
+                one.div(rate)?.add_checked(offset).map(|abs_mean| {
+                    if *positive_tail {
+                        Ok(abs_mean)
+                    } else {
+                        abs_mean.arithmetic_negate()
+                    }
+                })?
+            }
+            Gaussian { mean, .. } => Ok(mean.clone()),
+            Bernoulli { p } => Ok(p.clone()),
+            Unknown { mean, .. } => Ok(mean.clone()),
+        }
+    }
+
+    /// Extracts the median value of the given statistic, depending on its distribution:
+    /// - A [`Uniform`] distribution's interval determines its median value, which is the
+    ///   arithmetic average of the interval endpoints.
+    /// - An [`Exponential`] distribution's median is calculable by the formula `offset + ln2/λ`,
+    ///   where λ is the (non-negative) rate.
+    /// - A [`Gaussian`] distribution's median is equal to its mean, which is specified explicitly.
+    /// - A [`Bernoulli`] distribution's median is `1` if `p > 0.5` and `0` otherwise.
+    /// - An [`Unknown`] distribution _may_ have it explicitly, or this information may
+    ///   be absent.
+    pub fn median(&self) -> Result<ScalarValue> {
+        match &self {
+            Uniform { interval, .. } => {
+                let agg = interval
+                    .lower()
+                    .add_checked(interval.upper())?
+                    .cast_to(&DataType::Float64)?;
+                agg.div(ScalarValue::Float64(Some(2.)))
```

Comment:
```suggestion
                let dt = interval.lower.data_type();
                interval
                    .lower()
                    .add_checked(interval.upper())?
                    .div(ScalarValue::from(2).cast_to(dt))
```
This seems to polymorphize this section, it may be better to do it this way until we implement proper type promotion.